azure/Guidelines.md

```diff
@@ -0,0 +1,861 @@
+
+##### HTTP Return Codes
+> :white_check_mark: **DO** adhere to the return codes in the following table when the method is successful:
+
+Method | Description | Response Status Code 
+-------|-------------|---------------------
+PATCH  | Create/Modify the resource with JSON Merge Patch | 200-OK, 201-Created
+PUT    | Create/Replace the *whole* resource | 200-OK, 201-Created 
+POST   | Create new resource (ID set by service) | 201-Created with URL of created resource
+GET    | Read (i.e. list) a resource collection | 200-OK
+GET    | Read the resource | 200-OK 
+DELETE | Remove the resource | 204-No Content\; avoid 404-Not Found 
+ 
+> :white_check_mark: **DO** treat method names as case sensitive and should always be in uppercase
+> 
+> :white_check_mark: **DO** return the state of the resource after a PUT, PATCH, POST, or GET operation with a ```200-OK``` or ```201-Created```.
+> 
+> :white_check_mark: **DO** return a ```204-No Content``` without a resource/body for a DELETE operation (even if the URL identifies a resource that does not exist; do not return ```404-Not Found```)
+> 
+> :white_check_mark: **DO** support caching and optimistic concurrency by honoring the the if-match, if-none-match, if-modified-since, and if-unmodified-since request headers and by returning the etag and last-modified response headers
+
+### HTTP Query Parameters and Header Values
+Because information in the service URL, as well as the request / response, are strings, there must be a predictable, well-defined scheme to convert strings to their corresponding values.
+
+> :white_check_mark: **DO** validate all query parameter and request header values and fail the operation with ```400-Bad Request``` if any value fails validation. Return an error response as described in the [Handling Errors](#Handling-errors) section indicating what is wrong so customer can diagnose the issue and fix it themselves.
+> 
+> :white_check_mark: **DO** use the following table when translating strings:
+
+Data type | Document that string must be
+--------- | -------
+Boolean   | true / false (all lowercase)
+Integer   | -2<sup>53</sup>+1 to +2<sup>53</sup>-1 (for consistency with JSON limits on integers [RFC8259](https://datatracker.ietf.org/doc/html/rfc8259))
+Float     | [IEEE-754 binary64](https://en.wikipedia.org/wiki/Double-precision_floating-point_format)
+String    | (Un)quoted?, max length, legal characters, case-sensitive, multiple delimiter
+UUID      | 123e4567-e89b-12d3-a456-426614174000 (no {}s, hyphens, case-insensitive) [RFC4122](https://datatracker.ietf.org/doc/html/rfc4122)
+Date/Time (Header) | Sun, 06 Nov 1994 08:49:37 GMT [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.1)
+Date/Time (Query parameter) | YYYY-MM-DDTHH:mm:ss.sssZ [RFC3339](https://datatracker.ietf.org/doc/html/rfc3339)
```

Comment:
I would do a parenthesis like above to rewrite one more time "3 digits" so people can't miss it. Got too much hurt by this :)

Reply:
@lmazuel -- not sure what you are referring to on the parenthesis. however, we think you want us to emphasize that the time MUST be to three digits. We'll take a pass at this.